src/Tools/dotnet-counters/CounterMonitor.cs

```diff
@@ -190,22 +220,95 @@ private async Task<int> StartMonitor()
                 }
                 providerString = sb.ToString();
             }
+            return providerString;
+        }
+        
+
+        private async Task<int> Start(bool liveMonitor)
+        {
+            if (_processId == 0)
+            {
+                _console.Error.WriteLine("ProcessId is required.");
+                return 1;
+            }
+
+            // If we are exporting the counter data as file, do some more sanity checks.
+            if (!liveMonitor)
+            {
+                if (_output.Length == 0)
+                {
+                    _console.Error.WriteLine("Output cannot be an empty string");
+                    return 1;
+                }
+
+                if (_format == "csv")
+                {
+                    exporter = new CSVExporter();
+                }
+                else if (_format == "json")
+                {
+                    exporter = new JSONExporter();
+                }
+                else
+                {
+                    _console.Error.WriteLine($"The output format {_format} is not a valid output format.");
+                }
+
+                // Try getting the process name.
+                string processName = "";
+                try
+                {
+                    processName = Process.GetProcessById(_processId).ProcessName;
+                }
+                catch (Exception) {}
+                exporter.Initialize(_output, processName);
+            }
+
+            string providerString = BuildProviderString();
+            if (providerString.Length == 0)
+            {
+                return 1;
+            }
 
             ManualResetEvent shouldExit = new ManualResetEvent(false);
             _ct.Register(() => shouldExit.Set());
 
             var terminated = false;
-            writer.AssignRowsAndInitializeDisplay();
-
+            
+            if (liveMonitor)
+            {
+                writer.AssignRowsAndInitializeDisplay();
+            }
+            
             Task monitorTask = new Task(() => {
                 try
                 {
-                    RequestTracingV2(providerString);
+                    EventPipeEventSource source = RequestTracingV2(providerString);
+                    if (liveMonitor)
+                        source.Dynamic.All += DynamicAllMonitor;
+                    else
+                    {
+                        Debug.Assert(exporter != null, "exporter object should not be null if we are exporting");
+                        source.Dynamic.All += DynamicAllExport;
+
+                        Console.WriteLine("Starting a counter session. Press Q to quit.");
```

Comment:
This can become a call to renderer.EventPipeSourceConnected(), console writer has empty impl, file format writers have this Console.WriteLine here.